tests/unit/drivers/rank/test_matches2doc_rank_drivers.py

```diff
@@ -29,15 +29,9 @@ def __init__(self, *args, **kwargs):
             **kwargs
         )
 
-    def score(self, query_meta, old_match_scores, match_meta):
-        new_scores = [
-            (match_id, -abs(match_meta[match_id]['length'] - query_meta['length']))
-            for match_id, old_score in old_match_scores.items()
-        ]
-        return np.array(
-            new_scores,
-            dtype=[(self.COL_MATCH_ID, np.object), (self.COL_SCORE, np.float64)],
-        )
+    def score(self, old_match_scores, query_meta, match_meta):
+        new_scores = [-abs(m['length'] - query_meta['length']) for m in match_meta]
+        return new_scores
 
 
 def create_document_to_score():
```

Comment:
The way to generate data is a little bit confusing.

1. doc and match ID: Why doc.id = '1' * 20? In the comments the doc_id is '1' and the id for matches are 2,3,4,5. But in fact, we set them to match.id = str(match_id) * match_length; What's the reason?
2. Why `match.length = match_score`?

Reply:
1.the *20 responds to changes that have been applied to Document ID and continuous refactoring. 

2. I thinks is by test design, but this test is too complicated to be honest

Reply:
`match.score.value = match_score` this line seems doesn't work?

Reply:
> 1.the *20 responds to changes that have been applied to Document ID and continuous refactoring.
> 
> 1. I thinks is by test design, but this test is too complicated to be honest

😂

Reply:
where is this code?

Reply:
In the test, https://github.com/jina-ai/jina/blob/f0b6a44045f7fccca05a34020fd42981ce34dc4e/tests/unit/drivers/rank/test_matches2doc_rank_drivers.py#L61

Reply:
The value of the score cannot be set this way? In the test, this value is 0, thus old_matches_socres is [0.0,0.0,0.0,0.0]

Reply:
there maybe an error in this test, you are welcome to refactor

Reply:
> there maybe an error in this test, you are welcome to refactor

The refactor is in this PR, this makes more sense I think. We compare the abs diff between the length of matches with the query. Those with the same value will be sorted by id asending.
https://github.com/jina-ai/jina/pull/2127